### PR TITLE
autotest: Combine RFC35 tests

### PR DESCRIPTION
**Pull request by dbaston**
## What does this PR do?

Combines four test files that are mostly identical into a set of parameterized tests.

## Tasklist

 - [ ] Review
 - [ ] Adjust for comments
 - [ ] All CI builds and checks have passed

### CODE REVIEW

File: autotest/ogr/ogr_rfc35.py [hunk 1]
```diff
@@ -449,23 +504,25 @@ def test_ogr_rfc35_mitab_4():
...
+    else:
+        ds = ogr.Open(rfc35_test_input, update=1)
 
-    ds = ogr.Open("tmp/rfc35_test.tab", update=1)
     lyr = ds.GetLayer(0)
     lyr_defn = lyr.GetLayerDefn()
 
     assert lyr.TestCapability(ogr.OLCDeleteField) == 1
 
-    assert lyr.DeleteField(0) == 0
```
Comment (rouault) [inline autotest/ogr/ogr_rfc35.py:467]: is this removal intentional ?

Comment (dbaston) [reply autotest/ogr/ogr_rfc35.py:467]: Yes - in `master` the tests in this file run sequentially, so this line is cleaning up the addition of `oldintfld` near the end of the previous test. In this PR, each test gets a clean copy of the input dataset.
